src/interpreter-errors.ts

```diff
@@ -121,7 +127,7 @@ export class UnassignedVariable extends RuntimeSourceError {
   }
 
   public elaborate() {
-    return 'TODO'
+    return `Please assign a value to ${this.name} first. Example:\n\t${this.name} = 42;`
```

Comment:
This is wrong. Example of when this error occurs.

```javascript

const a = 1;
const f = () => {
  const i_think_i_can_access_outer_a_here = a; // UnassignedVariableError occurs here
  // other stuff;
  const a = 'another use of a';
}
f();
```

This is due to the proper scoping of variables, so access to any variables that are ever declared in the current scope is blocked until the statement where it is being declared is evaluated.